basic_system/src/system_implementation/system/da_commitment_generator/blob_commitment_generator/polynomial_evaluation.rs

```diff
@@ -0,0 +1,85 @@
+use super::brp_roots_of_unity::BRP_ROOTS_OF_UNITY;
+use super::*;
+use core::mem::MaybeUninit;
+
+///
+/// Evaluate blob polynomial in the given point.
+///
+/// Please note, that `data` is not the blob itself, but data we encode into the blob.
+/// For encoding, we chunk `data` by 31 bytes and interpret each chunk as BE blob element.
+///
+pub fn evaluate_blob_polynomial(data: &[u8], x: &crypto::bls12_381::Fr) -> crypto::bls12_381::Fr {
+    let mut poly: [MaybeUninit<crypto::bls12_381::Fr>; ELEMENTS_PER_4844_BLOCK] =
+        [MaybeUninit::uninit(); ELEMENTS_PER_4844_BLOCK];
+    let mut poly_iter = poly.iter_mut();
+
+    let chunks = data.array_chunks::<BLOB_CHUNK_SIZE>();
+    let mut last_chunk = [0u8; 31];
+    let remainder = chunks.remainder();
+    last_chunk[..remainder.len()].copy_from_slice(remainder);
+    for chunk in chunks {
+        poly_iter
+            .next()
+            .unwrap()
+            .write(crypto::bls12_381::Fr::from_bigint(parse_u256_be(chunk)).unwrap());
+    }
+    poly_iter
+        .next()
+        .unwrap()
+        .write(crypto::bls12_381::Fr::from_bigint(parse_u256_be(&last_chunk)).unwrap());
+    let poly = unsafe { MaybeUninit::array_assume_init(poly) };
```

Comment:
Also, is it ok to assume init? Should we 0 out the poly? Do we really need to use MaybeUninit here?